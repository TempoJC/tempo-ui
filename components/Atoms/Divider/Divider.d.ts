import { default as React } from '../../../../node_modules/react';
import { DividerProps } from './types';
/**
 * Use this component to show a divider.
 */
declare const Divider: React.FC<DividerProps>;
export default Divider;
