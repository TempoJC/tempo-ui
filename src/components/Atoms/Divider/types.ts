export const DIVIDER_DARK = "dark";
export const DIVIDER_LIGHT = "light";

export const DIVIDER_HORIZONTAL = "horizontal";
export const DIVIDER_VERTICAL = "vertical";

export const DIVIDER_CONTRAST_HIGH = "high";
export const DIVIDER_CONTRAST_MEDIUM = "medium";
export const DIVIDER_CONTRAST_LOW = "low";

export interface DividerProps {
  /**
   * Kind of the divider.
   */
  kind?: typeof DIVIDER_HORIZONTAL | typeof DIVIDER_VERTICAL;

  /**
   * Sets the color scheme of the component.
   */
  color?: typeof DIVIDER_LIGHT | typeof DIVIDER_DARK;

  /**
   * Contrast of the divider.
   */
  contrast?: typeof DIVIDER_CONTRAST_HIGH | typeof DIVIDER_CONTRAST_MEDIUM | typeof DIVIDER_CONTRAST_LOW;

  /**
   * Additional CSS to be applied to the component.
   */
  className?: string;

  /**
   * Adds a `data-testid` attribute to the component outermost wrapper, set to the given value.
   */
  testId?: string;
}
