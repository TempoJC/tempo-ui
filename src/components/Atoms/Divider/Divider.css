.tempo-divider {
  display: block;
}

.tempo-divider--horizontal {
  width: 100%;
  height: 1px;
}

.tempo-divider--vertical {
  width: 1px;
  height: 100%;
}

.tempo-divider--light.tempo-divider--high {
  background-color: var(--color-token-light-border-1);
}

.tempo-divider--dark.tempo-divider--high {
  background-color: var(--color-token-dark-border-1);
}

.tempo-divider--light.tempo-divider--medium {
  background-color: var(--color-token-light-border-2);
}

.tempo-divider--dark.tempo-divider--medium {
  background-color: var(--color-token-dark-border-2);
}

.tempo-divider--light.tempo-divider--low {
  background-color: var(--color-token-light-border-3);
}

.tempo-divider--dark.tempo-divider--low {
  background-color: var(--color-token-dark-border-3);
}
