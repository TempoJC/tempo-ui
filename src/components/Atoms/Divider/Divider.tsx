import React from "react";
import cn from "classnames";
import {
  DIVIDER_CONTRAST_HIGH,
  DIVIDER_CONTRAST_LOW,
  DIVIDER_CONTRAST_MEDIUM,
  DIVIDER_DARK,
  DIVIDER_HORIZONTAL,
  DIVIDER_LIGHT,
  DIVIDER_VERTICAL,
  DividerProps,
} from "./types";
import "./Divider.css";

/**
 * Use this component to show a divider.
 */
const Divider: React.FC<DividerProps> = ({
  className,
  kind = DIVIDER_HORIZONTAL,
  color = DIVIDER_LIGHT,
  contrast = DIVIDER_CONTRAST_HIGH,
  testId,
}) => (
  <div
    className={cn(
      className,
      "tempo-divider",
      {
        "tempo-divider--light": color === DIVIDER_LIGHT,
        "tempo-divider--dark": color === DIVIDER_DARK,
      },
      {
        "tempo-divider--horizontal": kind === DIVIDER_HORIZONTAL,
        "tempo-divider--vertical": kind === DIVIDER_VERTICAL,
      },
      {
        "tempo-divider--high": contrast === DIVIDER_CONTRAST_HIGH,
        "tempo-divider--medium": contrast === DIVIDER_CONTRAST_MEDIUM,
        "tempo-divider--low": contrast === DIVIDER_CONTRAST_LOW,
      },
    )}
    data-testid={testId}
  />
);

export default Divider;
