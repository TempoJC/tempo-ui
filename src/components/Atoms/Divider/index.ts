// packages/components-divider/divider
export { default as Divider } from "./Divider";
export * from "./types";
