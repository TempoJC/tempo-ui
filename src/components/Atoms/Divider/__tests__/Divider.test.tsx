import { render } from "@testing-library/react";
import Divider from "../Divider";

describe("@tempo-ui/Atoms/Divider", () => {
  it("renders", () => {
    const { asFragment } = render(<Divider />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe("kind", () => {
    it("is horizontal", () => {
      const { asFragment } = render(<Divider kind="horizontal" />);
      expect(asFragment()).toMatchSnapshot();
    });
    it("is vertical", () => {
      const { asFragment } = render(<Divider kind="vertical" />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("contrast & color", () => {
    it("is contrast high and color light", () => {
      const { asFragment } = render(<Divider />);
      expect(asFragment()).toMatchSnapshot();
    });
    it("is contrast medium and color light", () => {
      const { asFragment } = render(<Divider contrast="medium" />);
      expect(asFragment()).toMatchSnapshot();
    });
    it("is contrast low and color light", () => {
      const { asFragment } = render(<Divider contrast="low" />);
      expect(asFragment()).toMatchSnapshot();
    });
    it("is contrast high and color dark", () => {
      const { asFragment } = render(<Divider color="dark" />);
      expect(asFragment()).toMatchSnapshot();
    });
    it("is contrast medium and color dark", () => {
      const { asFragment } = render(<Divider contrast="medium" color="dark" />);
      expect(asFragment()).toMatchSnapshot();
    });
    it("is contrast low and color dark", () => {
      const { asFragment } = render(<Divider contrast="low" color="dark" />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it("adds the additional className", () => {
    const { container } = render(<Divider className="customClass" />);
    expect(container.querySelector(".customClass")).toBeInTheDocument();
  });

  describe("testId prop", () => {
    it("renders the divider with the specified data-testid attribute", () => {
      const { getByTestId } = render(<Divider testId="divider" />);

      const element = getByTestId("divider");

      expect(element).toBeInTheDocument();
      expect(element).toHaveAttribute("data-testid", "divider");
    });
  });
});
