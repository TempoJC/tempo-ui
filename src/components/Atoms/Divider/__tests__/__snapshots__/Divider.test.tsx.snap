// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`@tempo-ui/Atoms/Divider contrast & color is contrast high and color dark 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--dark tempo-divider--horizontal tempo-divider--high"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider contrast & color is contrast high and color light 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--light tempo-divider--horizontal tempo-divider--high"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider contrast & color is contrast low and color dark 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--dark tempo-divider--horizontal tempo-divider--low"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider contrast & color is contrast low and color light 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--light tempo-divider--horizontal tempo-divider--low"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider contrast & color is contrast medium and color dark 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--dark tempo-divider--horizontal tempo-divider--medium"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider contrast & color is contrast medium and color light 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--light tempo-divider--horizontal tempo-divider--medium"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider kind is horizontal 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--light tempo-divider--horizontal tempo-divider--high"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider kind is vertical 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--light tempo-divider--vertical tempo-divider--high"
  />
</DocumentFragment>
`;

exports[`@tempo-ui/Atoms/Divider renders 1`] = `
<DocumentFragment>
  <div
    class="tempo-divider tempo-divider--light tempo-divider--horizontal tempo-divider--high"
  />
</DocumentFragment>
`;
